File: src/main/resources/static/src/_model/team.ts
import { Person } from './person';
export class Team {
    private _id?: number;
    private _name?: string;
    private _authorities?: string;
    private _persons?: Person[];

    /**
    * Getter id
    * @return {number}
    */
    public get id(): number {
        return this._id;
    }

    /**
     * Getter name
     * @return {string}
     */
    public get name(): string {
        return this._name;
    }

    /**
     * Getter authorities
     * @return {string}
     */
    public get authorities(): string {
        return this._authorities;
    }

    /**
     * Getter persons
     * @return {Person[]}
     */
    public get persons(): Person[] {
        return this._persons;
    }

    /**
     * Setter id
     * @param {number} value
     */
    public set id(value: number) {
        this._id = value;
    }

    /**
     * Setter name
     * @param {string} value
     */
    public set name(value: string) {
        this._name = value;
    }

    /**
     * Setter authorities
     * @param {string} value
     */
    public set authorities(value: string) {
        this._authorities = value;
    }

    /**
     * Setter persons
     * @param {Person[]} value
     */
    public set persons(value: Person[]) {
        this._persons = value;
    }

}

File: src/main/java/com/besafx/app/ink/model/Team.java
package com.besafx.app.ink.model;

import com.besafx.app.ink.config.Auditable;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.OneToMany;
import java.io.IOException;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@JsonIgnoreProperties(ignoreUnknown = true)
public class Team extends Auditable<String> implements Serializable {

    private static final long serialVersionUID = 1L;

    @GenericGenerator(
            name = "teamSequenceGenerator",
            strategy = "org.hibernate.id.enhanced.SequenceStyleGenerator",
            parameters = {
                    @org.hibernate.annotations.Parameter(name = "sequence_name", value = "TEAM_SEQUENCE"),
                    @org.hibernate.annotations.Parameter(name = "initial_value", value = "1"),
                    @org.hibernate.annotations.Parameter(name = "increment_size", value = "1")
            }
    )
    @Id
    @GeneratedValue(generator = "teamSequenceGenerator")
    private Long id;

    private String name;

    @Lob
    @Type(type = "org.hibernate.type.TextType")
    private String authorities;

    @OneToMany(mappedBy = "team")
    private List<Person> persons = new ArrayList<>();

    @JsonCreator
    public static Team Create(String jsonString) throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        Team team = mapper.readValue(jsonString, Team.class);
        return team;
    }
}
